app/controllers/api/grades_controller.rb

```diff
@@ -0,0 +1,49 @@
+module Api
+  class GradesController < ApiController
+    skip_before_action :doorkeeper_authorize!
+
+    def orca_response
+      body = orca_response_params
+      key = body[:key]
+      grade_id = key['grade_id']
+      secret = key['secret']
+      grade = Grade.find_by(id: grade_id)
+      if grade.nil?
+        return render json: { message: "No grade for ID #{grade_id}" },
+                      status: :not_found
+      end
+      grader_dir = grade.submission_grader_dir
+      secret_file_path = grader_dir.join('orca.secret')
```

Comment:
Ask the grader for its secret, don't hard-code a magic constant here.  `grade.grader.orca_secret_path` would be appropriate.